KG-315 Exclude Ktor engine dependencies

kpavlov opened the pull request:
## Motivation and Context

KG-315 / #710 To avoid Ktor CIO dependency leak, it was excluded from API dependencies in all modules.
Client and server Ktor engine dependencies should be manually included in the end application and tests.

## Changes

- Removed platform-specific `HttpClientEngineFactory` implementations from the `prompt-executor-ollama-client` module.
- Consolidated `HttpClient` initialization using the default engine.
- Adjusted build configurations to reallocate `ktor-client-cio` dependencies under appropriate test source sets.
- Updated relevant documentation and examples to explicitly include engine dependencies where necessary.
- Rename deprecated onAgentFinished to onAgentCompleted across tests
- Updated dependencies in example: add ktor client dependencies explicitly 

## Breaking Changes
Client and server Ktor engine dependencies should now be manually included.

---

#### Type of the changes
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update
- [x] Tests improvement
- [x] Refactoring

#### Checklist
- [x] The pull request has a description of the proposed change
- [x] I read the [Contributing Guidelines](https://github.com/JetBrains/koog/blob/main/CONTRIBUTING.md) before opening the pull request
- [x] The pull request uses **`develop`** as the base branch
- [ ] Tests for the changes have been added
- [x] All new and existing tests passed

##### Additional steps for pull requests adding a new feature
- [ ] An issue describing the proposed change exists
- [ ] The pull request includes a link to the issue
- [ ] The change was discussed and approved in the issue
- [ ] Docs have been added / updated

kpavlov committed:
Update dependencies in `spring-boot-java` example

- Bump Spring Boot version to `3.5.6`.
- Update Ktor version to `3.2.3`.
- Adjust Koog version to `0.4.3-develop-20250926-0205`.

kpavlov committed:
Knit fix

kpavlov commented on agents/agents-core/src/commonTest/kotlin/ai/koog/agents/core/dsl/extension/AIAgentNodesHistoryCompressionTest.kt:
```diff
@@ -64,7 +64,7 @@ class AIAgentNodesHistoryCompressionTest {
         ) {
             install(EventHandler) {
-                onAgentFinished { eventContext -> results += eventContext.result }
+                onAgentCompleted { eventContext -> results += eventContext.result }
```
here and other places like that: Replace the deprecated api method

kpavlov commented on agents/agents-features/agents-features-snapshot/src/jvmTest/kotlin/PersistencyTestUtils.kt:
```diff
@@ -1,7 +1,16 @@
 
 internal class TestAgentLogsCollector {
-    val logs = ConcurrentLinkedQueue<String>()
+    private val logs = LinkedBlockingQueue<String>()
```
encapsulate logs, use a thread-safe linked list with atomic clear operation

kpavlov commented on prompt/prompt-executor/prompt-executor-clients/prompt-executor-ollama-client/src/commonMain/kotlin/ai/koog/prompt/executor/ollama/client/OllamaClient.kt:
```diff
@@ -426,5 +425,3 @@ public class OllamaClient(
     }
 }
-
-internal expect fun engineFactoryProvider(): HttpClientEngineFactory<*>
```
Manual engine creation is no longer needed: EmbeddedServer can auto-detect it

devcrocod commented on prompt/prompt-executor/prompt-executor-clients/prompt-executor-ollama-client/build.gradle.kts:
```diff
@@ -30,33 +30,27 @@ kotlin {
+        appleTest {
             dependencies {
-                api(libs.ktor.client.darwin)
+                implementation(libs.ktor.client.darwin)
```
Shouldn’t the same be done for the other clients as well?

devcrocod reviewed (approved):
I have a couple of small questions,
but other than that it looks good to me

kpavlov committed:
Reallocate `ktor-client.js` dependencies under `jsTest` source set and remove unused `ktor-client-cio` dependency.

kpavlov commented on agents/agents-core/build.gradle.kts:
```diff
@@ -47,6 +47,7 @@ kotlin {
             dependencies {
                 implementation(kotlin("test-junit5"))
                 implementation("org.jetbrains.lincheck:lincheck:3.1")
+                implementation(libs.ktor.client.cio)
```
It is actually required in `FeatureMessageRemoteClientTest`

kpavlov committed:
Add `ktor-client-darwin` dependency to `appleTest` configuration

EugeneTheDev committed:
Remove platform-specific Ktor engine dependencies and clean up unused code

- Removed platform-specific `HttpClientEngineFactory` implementations from the `prompt-executor-ollama-client` module.
- Consolidated `HttpClient` initialization using the default engine.
- Adjusted build configurations to reallocate `ktor-client-cio` dependencies under appropriate test source sets.
- Updated relevant documentation and examples to explicitly include engine dependencies where necessary.

# Conflicts:
#	agents/agents-core/build.gradle.kts

EugeneTheDev committed:
Add `ktor-client-cio` dependency to test configuration

# Conflicts:
#	prompt/prompt-executor/prompt-executor-clients/prompt-executor-openai-client/build.gradle.kts

EugeneTheDev committed:
Examples: Add dependencies for Ktor clients across modules and update Koog version

- Introduced platform-specific Ktor client dependencies (`cio`, `okhttp`, `apache5`, `darwin`, `js`).
- Added `ktor-bom` for dependency management.
- Updated Koog version to `0.4.3-develop-20250927-0204`.

# Conflicts:
#	examples/demo-compose-app/gradle/libs.versions.toml

EugeneTheDev committed:
[koog-agents] Add per-platform Ktor clients
